pandas/tests/io/test_stata.py

```diff
@@ -1327,3 +1329,23 @@ def test_set_index(self):
             df.to_stata(path)
             reread = pd.read_stata(path, index_col='index')
         tm.assert_frame_equal(df, reread)
+
+    @pytest.mark.parametrize(
+        'column', ['ms', 'day', 'week', 'month', 'qtr', 'half', 'yr'])
+    def test_date_parsing_ignores_format_details(self, column):
+        """
+        Test that display formats are ignored when determining if a numeric
+        column is a date value.
+
+        All date types are stored as numbers and format associated with the
+        column denotes both the type of the date and the display format.
+
+        STATA supports 9 date types which each have distinct units. We test 7
+        of the 9 types, ignoring %tC and %tb. %tC is a variant of %tc that
+        accounts for leap seconds and %tb relies on STATAs business calendar.
+        """
```

Comment:
Great description!  Can you convert that to a comment block.  That's more aligned with how we comment our tests, and also, reference the issue number at the top (instead of at the bottom).